File: Euclid/Messages/Outgoing/WHISPER.cs
using Euclid.Network.Streams.Util;

namespace Euclid.Messages.Outgoing
{
    class WHISPER : IMessageComposer
    {
        private string name;
        private string chatMsg;

        public WHISPER(string name, string chatMsg)
        {
            this.name = name;
            this.chatMsg = chatMsg;
        }

        public override void Write()
        {
            this.Data.Add(new ArgumentEntry(this.name));
            this.Data.Add(new DelimeterEntry(this.chatMsg, ' '));
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/WHISPER.java
package euclid.messages.outgoing;

import euclid.messages.IMessageComposer;
import euclid.network.streams.util.ArgumentEntry;
import euclid.network.streams.util.DelimeterEntry;

public class WHISPER extends IMessageComposer {

    private final String name;
    private final String chatMsg;

    public WHISPER(String name, String chatMsg) {
        this.name = name;
        this.chatMsg = chatMsg;
    }

    @Override
    public void write() {
        getData().add(new ArgumentEntry(name));
        getData().add(new DelimeterEntry(chatMsg, " "));
    }
}
